sagemaker-spark-sdk/src/main/scala/com/amazonaws/services/sagemaker/sparksdk/algorithms/FactorizationMachinesSageMakerEstimator.scala

```diff
@@ -0,0 +1,779 @@
+
+/**
+  * Common params for [[FactorizationMachinesSageMakerEstimator]] with accessors
+  */
+private[algorithms] trait FactorizationMachinesParams extends SageMakerAlgorithmParams {
+
+  /**
+    * Dimensionality of factorization. Must be > 0.
+    * Required
+    */
+  val numFactors : IntParam = new IntParam(this, "num_factors",
+    "Dimensionality of factorization. Must be > 0.",
+    ParamValidators.gt(0))
+  def getNumFactors: Int = $(numFactors)
+
+  /**
+    * Whether training is for binary classification or regression.
+    * Supported options: "binary_classifier", and "regressor".
+    * Required
+    */
+  private[algorithms] val predictorType : Param[String] = new Param(this, "predictor_type",
+    "Whether training is for binary classification or regression. " +
+      "Supported options: 'binary_classifier', and 'regressor'.",
+    ParamValidators.inArray(Array("binary_classifier", "regressor")))
+
+  /**
+    * Number of training epochs to run. Must be > 0.
+    * Default: 1
+    */
+  val epochs : IntParam = new IntParam(this, "epochs",
+    "Number of training epochs to run. Must be > 0.",
+    ParamValidators.gt(0))
+  def getEpochs: Int = $(epochs)
+
+  /**
+    * Clip the gradient by projecting onto the box [-clip_gradient, +clip_gradient].
+    */
+  val clipGradient : DoubleParam = new DoubleParam(this, "clip_gradient",
+    "Clip the gradient by projecting onto the box [-clip_gradient, +clip_gradient].")
+  def getClipGradient: Double = $(clipGradient)
+
+  /**
+    * Small value to avoid division by 0.
+    */
+  val eps : DoubleParam = new DoubleParam(this, "eps",
+    "Small value to avoid division by 0.")
+  def getEps: Double = $(eps)
+
+  /**
+    * If set, multiplies the gradient with rescale_grad before updating.
+    * Often choose to be 1.0/batch_size.
+    */
+  val rescaleGrad : DoubleParam = new DoubleParam(this, "rescale_grad",
+    "Multiplies the gradient with this value before updating")
+  def getRescaleGrad: Double = $(rescaleGrad)
+
+  /**
+    * Non-negative learning rate for the bias term.
+    * Suggested value range: [1e-8, 512].
+    * Default: 0.1
+    */
+  val biasLr : DoubleParam = new DoubleParam(this, "bias_lr",
+    "Multiplies the gradient with this value before updating. Must be > 0",
+    ParamValidators.gt(0))
+  def getBiasLr: Double = $(biasLr)
+
+  /**
+    * Non-negative learning rate for linear terms.
+    * Suggested value range: [1e-8, 512].
+    * Default: 0.001
+    */
+  val linearLr : DoubleParam = new DoubleParam(this, "linear_lr",
+    "Learning rate for linear terms. Must be > 0",
+    ParamValidators.gt(0))
+  def getLinearLr: Double = $(linearLr)
+
+  /**
+    * Non-negative learning rate for factorization terms.
+    * Suggested value range: [1e-8, 512].
+    * Default: 0.0001
+    */
+  val factorsLr : DoubleParam = new DoubleParam(this, "factors_lr",
+    "Learning rate for factorization terms. Must be > 0",
+    ParamValidators.gt(0))
```

Comment:
The FM documentation says "non-negative float" for many of these hyperparams: https://docs.aws.amazon.com/sagemaker/latest/dg/fact-machines-hyperparameters.html

Can you change these to `ParamValidators.gtEq(0)`?

Reply:
Thanks - will change.